src/masternodes/mn_checks.cpp

```diff
@@ -5004,19 +4992,23 @@ bool IsMainNetwork() {
     return Params().NetworkIDString() == CBaseChainParams::MAIN;
 }
 
-bool IsICXEnabled(const int height, const CCustomCSView &view) {
-    if (height < Params().GetConsensus().NextNetworkUpgradeHeight) {
-        return true;
+bool IsICXEnabled(const int height, const CCustomCSView &view, const Consensus::Params &consensus) {
+    if (height >= consensus.NextNetworkUpgradeHeight) {
+        const CDataStructureV0 enabledKey{AttributeTypes::Param, ParamIDs::Feature, DFIPKeys::ICXEnabled};
+        auto attributes = view.GetAttributes();
+        assert(attributes);
+        return attributes->GetValue(enabledKey, false);
     }
-
-    const CDataStructureV0 enabledKey{AttributeTypes::Param, ParamIDs::Feature, DFIPKeys::ICXEnabled};
-    auto attributes = view.GetAttributes();
-    assert(attributes);
-    return attributes->GetValue(enabledKey, false);
+    // ICX transactions allowed before NextNetwrokUpgrade and some of these conditions
+    else if (height < consensus.FortCanningParkHeight || IsRegtestNetwork() || (IsTestNetwork() && static_cast<int>(height) >= 1250000))
+        return true;
+    // ICX transactions disabled
+    else if (height >= consensus.FortCanningParkHeight && height < consensus.NextNetworkUpgradeHeight)
```

Comment:
Looks like you can just skip this else if and return false. Thinking this function might generate a warning about no return at the end.

Reply:
It does indeed generate this warning.

```
masternodes/mn_checks.cpp: In function ‘bool IsICXEnabled(int, const CCustomCSView&, const Consensus::Params&)’:
masternodes/mn_checks.cpp:5014:1: warning: control reaches end of non-void function [-Wreturn-type]
```
In addition to that there are  these warnings as well.

```
masternodes/mn_checks.cpp: In lambda function:
masternodes/mn_checks.cpp:2406:48: warning: comparison of integer expressions of different signedness: ‘uint32_t’ {aka ‘unsigned int’} and ‘const int’ [-Wsign-compare]
 2406 |                     (IsMainNetwork() && height >= consensus.NextNetworkUpgradeHeight))
      |                                         ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
masternodes/mn_checks.cpp: In function ‘bool IsDisabledTx(uint32_t, CustomTxType, const Consensus::Params&)’:
masternodes/mn_checks.cpp:4037:10: warning: unused variable ‘nextNetworkUpgradeHeight’ [-Wunused-variable]
 4037 |     auto nextNetworkUpgradeHeight = static_cast<uint32_t>(consensus.NextNetworkUpgradeHeight);
      |          ^~~~~~~~~~~~~~~~~~~~~~~~
```

Reply:
Thnx for the review, resolved!